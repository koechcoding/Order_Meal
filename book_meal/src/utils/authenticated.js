export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}