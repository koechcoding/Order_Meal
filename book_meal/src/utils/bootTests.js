import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
