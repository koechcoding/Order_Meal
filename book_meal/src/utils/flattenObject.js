export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}