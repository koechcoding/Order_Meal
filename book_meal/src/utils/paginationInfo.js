export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}