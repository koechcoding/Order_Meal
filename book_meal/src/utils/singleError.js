const singleError = (errorPayload)=>{
    if(errorPayload && errorPayload.data){
        const response = errorPayload.data;
        //alternative to Object.keys...
        for(let errorKey in response.errors){
            return response.error[errorKey]
        }
        if(response.message){
            return response.message;
        }
    }
    return 'Unknown error. Please try again!';
}
export default singleError;