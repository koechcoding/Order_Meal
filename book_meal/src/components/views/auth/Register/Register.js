import React from 'react';
import { Alert, Button } from 'reactstrap';
import { Link } from 'react-router-dom';
import axios from 'src/axios';

import AuthPage from '../../../common/AuthPage';
import { singleError } from '../../../../utils';

class Register extends React.Component {

    state = {}

    onSubmit = (e) => {
        e.preventDefault();
        this.setState({
            ...this.state,
            loading: true,
        })
        axios.post('auth/signup', this.state).then(({ data }) => {
            this.props.history.push('/account-created');
        }).catch(({ response }) => {
            this.setState({
                ...this.state,
                error: response,
                loading: false,
            })
        })
    }

    onChange = (e) => {
        this.setState({
            ...this.state,
            [e.target.name]: e.target.value
        })
    }

    render() {

        const { loading, error } = this.state;

        return (
            <AuthPage styles={{ minHeight: '780px', height: '100vh' }} loading={loading}>
                <form 
                    className="col-12 col-md-6 offset-md-3 col-lg-4 offset-lg-4 card"
                    onSubmit={this.onSubmit}>

                    <h5 className="text-center mb-3 mt-4">Create An Account</h5>
                    {error && <Alert color="danger"> {singleError(error)} </Alert>}
                    <label>Username</label>
                    <input 
                        className="form-control" 
                        type="text" 
                        name="username" 
                        onChange={this.onChange} />
                    <label>Email</label>
                    <input 
                        className="form-control" 
                        type="email" 
                        name="email" 
                        onChange={this.onChange} />
                    <label>Password </label>
                    <input 
                        className="form-control" 
                        type="password" 
                        name="password" 
                        onChange={this.onChange} />
                    <label>Confirm Password</label>
                    <input 
                        className="form-control" 
                        type="password" 
                        name="password_confirmation" 
                        onChange={this.onChange} />
                    <Button disabled={loading} className="btn btn-primary mt-2">Sign Up</Button>
                    <p className="text-center pt-3"> 
                        <Link to="/login">Login?</Link>
                    </p>
                </form>
            </AuthPage>
        );
    }
}

export default Register;