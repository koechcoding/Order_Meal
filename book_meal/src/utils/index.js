import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}