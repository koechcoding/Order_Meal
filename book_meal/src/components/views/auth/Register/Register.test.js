import React from "react";
import { shallow } from 'enzyme';
import 'src/utils/bootTests';
import Register  from "./Register";

describe('<Register />', ()=>{
    it('renders correctly', ()=>{
        const wrapper = shallow(<Register/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('records changes', ()=>{
        const wrapper = shallow(<Register/>);
        wrapper.instance().onChange({target:{name: 'name', value: 'value'}});
        expect(wrapper.state('name')).toBe('value');
    });
});