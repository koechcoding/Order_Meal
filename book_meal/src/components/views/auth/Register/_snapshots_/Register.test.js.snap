// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <form
        className="col-12 col-md-6 offset-md-3 col-lg-4 offset-lg-4 card"
        onSubmit={[Function]}
>
        <h5
                className="text-center mb-3 mt-4"
        >
                Create An Account
        </h5>
        <label>
                Username
        </label>
        <input
                className="form-control"
                name="username"
                onChange={[Function]}
                type="text"
        />
        <label>
                Email
        </label>
        <input
                className="form-control"
                name="email"
                onChange={[Function]}
                type="email"
        />
        <label>
                Password 
        </label>
        <input
                className="form-control"
                name="password"
                onChange={[Function]}
                type="password"
        />
        <label>
                Confirm Password
        </label>
        <input
                className="form-control"
                name="password_confirmation"
                onChange={[Function]}
                type="password"
        />
        <Button
                className="btn btn-primary mt-2"
                color="secondary"
                disabled={undefined}
                tag="button"
        >
                Sign Up
        </Button>
        <p
                className="text-center pt-3"
        >
                <Link
                        replace={false}
                        to="/login"
                >
                        Login?
                </Link>
        </p>
</form>,
      "loading": undefined,
      "styles": Object {
        "height": "100vh",
        "minHeight": "780px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5
            className="text-center mb-3 mt-4"
>
            Create An Account
</h5>,
          undefined,
          <label>
            Username
</label>,
          <input
            className="form-control"
            name="username"
            onChange={[Function]}
            type="text"
/>,
          <label>
            Email
</label>,
          <input
            className="form-control"
            name="email"
            onChange={[Function]}
            type="email"
/>,
          <label>
            Password 
</label>,
          <input
            className="form-control"
            name="password"
            onChange={[Function]}
            type="password"
/>,
          <label>
            Confirm Password
</label>,
          <input
            className="form-control"
            name="password_confirmation"
            onChange={[Function]}
            type="password"
/>,
          <Button
            className="btn btn-primary mt-2"
            color="secondary"
            disabled={undefined}
            tag="button"
>
            Sign Up
</Button>,
          <p
            className="text-center pt-3"
>
            <Link
                        replace={false}
                        to="/login"
            >
                        Login?
            </Link>
</p>,
        ],
        "className": "col-12 col-md-6 offset-md-3 col-lg-4 offset-lg-4 card",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create An Account",
            "className": "text-center mb-3 mt-4",
          },
          "ref": null,
          "rendered": "Create An Account",
          "type": "h5",
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Username",
          },
          "ref": null,
          "rendered": "Username",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "name": "username",
            "onChange": [Function],
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Email",
          },
          "ref": null,
          "rendered": "Email",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "name": "email",
            "onChange": [Function],
            "type": "email",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Password ",
          },
          "ref": null,
          "rendered": "Password ",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "name": "password",
            "onChange": [Function],
            "type": "password",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Confirm Password",
          },
          "ref": null,
          "rendered": "Confirm Password",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "name": "password_confirmation",
            "onChange": [Function],
            "type": "password",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Sign Up",
            "className": "btn btn-primary mt-2",
            "color": "secondary",
            "disabled": undefined,
            "tag": "button",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/login"
>
              Login?
</Link>,
            "className": "text-center pt-3",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Login?",
              "replace": false,
              "to": "/login",
            },
            "ref": null,
            "rendered": "Login?",
            "type": [Function],
          },
          "type": "p",
        },
      ],
      "type": "form",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <form
          className="col-12 col-md-6 offset-md-3 col-lg-4 offset-lg-4 card"
          onSubmit={[Function]}
>
          <h5
                    className="text-center mb-3 mt-4"
          >
                    Create An Account
          </h5>
          <label>
                    Username
          </label>
          <input
                    className="form-control"
                    name="username"
                    onChange={[Function]}
                    type="text"
          />
          <label>
                    Email
          </label>
          <input
                    className="form-control"
                    name="email"
                    onChange={[Function]}
                    type="email"
          />
          <label>
                    Password 
          </label>
          <input
                    className="form-control"
                    name="password"
                    onChange={[Function]}
                    type="password"
          />
          <label>
                    Confirm Password
          </label>
          <input
                    className="form-control"
                    name="password_confirmation"
                    onChange={[Function]}
                    type="password"
          />
          <Button
                    className="btn btn-primary mt-2"
                    color="secondary"
                    disabled={undefined}
                    tag="button"
          >
                    Sign Up
          </Button>
          <p
                    className="text-center pt-3"
          >
                    <Link
                              replace={false}
                              to="/login"
                    >
                              Login?
                    </Link>
          </p>
</form>,
        "loading": undefined,
        "styles": Object {
          "height": "100vh",
          "minHeight": "780px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h5
              className="text-center mb-3 mt-4"
>
              Create An Account
</h5>,
            undefined,
            <label>
              Username
</label>,
            <input
              className="form-control"
              name="username"
              onChange={[Function]}
              type="text"
/>,
            <label>
              Email
</label>,
            <input
              className="form-control"
              name="email"
              onChange={[Function]}
              type="email"
/>,
            <label>
              Password 
</label>,
            <input
              className="form-control"
              name="password"
              onChange={[Function]}
              type="password"
/>,
            <label>
              Confirm Password
</label>,
            <input
              className="form-control"
              name="password_confirmation"
              onChange={[Function]}
              type="password"
/>,
            <Button
              className="btn btn-primary mt-2"
              color="secondary"
              disabled={undefined}
              tag="button"
>
              Sign Up
</Button>,
            <p
              className="text-center pt-3"
>
              <Link
                            replace={false}
                            to="/login"
              >
                            Login?
              </Link>
</p>,
          ],
          "className": "col-12 col-md-6 offset-md-3 col-lg-4 offset-lg-4 card",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create An Account",
              "className": "text-center mb-3 mt-4",
            },
            "ref": null,
            "rendered": "Create An Account",
            "type": "h5",
          },
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Username",
            },
            "ref": null,
            "rendered": "Username",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "name": "username",
              "onChange": [Function],
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Email",
            },
            "ref": null,
            "rendered": "Email",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "name": "email",
              "onChange": [Function],
              "type": "email",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Password ",
            },
            "ref": null,
            "rendered": "Password ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "name": "password",
              "onChange": [Function],
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Confirm Password",
            },
            "ref": null,
            "rendered": "Confirm Password",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "name": "password_confirmation",
              "onChange": [Function],
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Sign Up",
              "className": "btn btn-primary mt-2",
              "color": "secondary",
              "disabled": undefined,
              "tag": "button",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/login"
>
                Login?
</Link>,
              "className": "text-center pt-3",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Login?",
                "replace": false,
                "to": "/login",
              },
              "ref": null,
              "rendered": "Login?",
              "type": [Function],
            },
            "type": "p",
          },
        ],
        "type": "form",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;